node/src/main/java/ch/prevo/open/node/services/JobEndService.java

```diff
@@ -1,22 +1,28 @@
 package ch.prevo.open.node.services;
 
-import java.util.Arrays;
-import java.util.HashSet;
 import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
 
 import org.springframework.stereotype.Service;
 
 import ch.prevo.open.encrypted.model.InsurantInformation;
+import ch.prevo.open.node.data.provider.JobEndProvider;
 
 /**
- * Dummy mock service implementation to retrieve encrypted information about an insurant.
+ * Service implementation to retrieve encrypted information about an insurant.
  */
 @Service
 public class JobEndService {
 
-    public Set<InsurantInformation> getAllJobEndData() {
-        return new HashSet<>(Arrays.asList(new InsurantInformation("756.3412.8844.97", "CHE-109.537.488"),
-                new InsurantInformation("756.1335.5778.23", "CHE-109.740.084"),
-                new InsurantInformation("756.9534.5271.94", "CHE-109.740.078")));
-    }
+	@Inject
+	private JobEndProvider jobEndProvider;
```

Comment:
Ich würde es bevorzugen, wenn wir nur Constructor Injection nutzen, da Field Injection eine Reihe von Nachteilen hat.

Reply:
Was sind genau die Nachteilen?

Reply:
Hier sind ein paar Gründe, die mir mal spontan einfallen:
- Die Felder können nicht final sein.
- Du hast viel bessere Kontrolle in Tests, was du dort injecten möchtest. Vielleicht willst du ein paar Mocks selber schreiben, ein paar von einem Mocking Framework generieren und ein paar Eingabeparameter tatsächlich anlegen.
- Wenn etwas schief geht, lässt sich Constructor Injection einfacher analysieren, indem du eine Breakpoint im Constructor setzt.